### PR TITLE
fix ncll type & use_cache

### PR DESCRIPTION
**Pull request by chiphuyen**
1. convert masks from bool to int to work with pytorch >= 1.3
2. change typo self.use_cache to use_cache

Signed-off-by: Chip Nguyen <huyenntkvn@gmail.com>

### CODE REVIEW

File: examples/nlp/joint_intent_slot_infer.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 parser.add_argument("--dataset_name", default='snips-all', type=str)
 parser.add_argument("--data_dir", default='data/nlu/snips', type=str)
 parser.add_argument("--work_dir",
-                    default='outputs/SNIPS-ALL/20191014-104316/checkpoints',
+                    default='outputs/SNIPS-ALL/2019-12-03_17-23-21/checkpoints',
```
Comment (blisc) [inline examples/nlp/joint_intent_slot_infer.py]: This is a poor default.

File: examples/nlp/joint_intent_slot_infer_b1.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
                     type=str)
 parser.add_argument("--query", default='please turn on the light', type=str)
 parser.add_argument("--work_dir",
-                    default='outputs/SNIPS-ALL/20191010-164934/checkpoints',
+                    default='outputs/SNIPS-ALL/2019-12-03_17-23-21/checkpoints',
```
Comment (blisc) [inline examples/nlp/joint_intent_slot_infer_b1.py]: This is a poor default.

**Comment by blisc** [comment]
./examples/nlp/joint_intent_slot_infer.py:24:80: E501 line too long (80 > 79 characters)
./examples/nlp/joint_intent_slot_infer_b1.py:26:80: E501 line too long (80 > 79 characters)

so close yet a little bit too much :D

**Comment by chiphuyen** [comment]
> ./examples/nlp/joint_intent_slot_infer.py:24:80: E501 line too long (80 > 79 characters)
> ./examples/nlp/joint_intent_slot_infer_b1.py:26:80: E501 line too long (80 > 79 characters)
> 
> so close yet a little bit too much :D

classic blunder

File: collections/nemo_nlp/nemo_nlp/data/datasets/joint_intent_slot.py [hunk 3]
```diff
@@ -57,27 +57,27 @@ def get_features(queries,
         words = query.strip().split()
         subtokens = ['[CLS]']
         loss_mask = [not ignore_start_end]
```
Comment (ekmb) [inline collections/nemo_nlp/nemo_nlp/data/datasets/joint_intent_slot.py]: loss_mask = [not ignore_start_end] should be  [int(not ignore_start_end)]

File: collections/nemo_nlp/nemo_nlp/data/datasets/joint_intent_slot.py [hunk 4]
```diff
@@ -98,7 +98,7 @@ def get_features(queries,
             all_input_mask[i] = [1] + all_input_mask[i][-max_seq_length + 1:]
             all_loss_mask[i] = [not ignore_start_end] + \
```
Comment (ekmb) [inline collections/nemo_nlp/nemo_nlp/data/datasets/joint_intent_slot.py]: and here also  should be  [int(not ignore_start_end)]

File: examples/nlp/joint_intent_slot_infer.py (hunk 1 above)
Comment (blisc) [reply examples/nlp/joint_intent_slot_infer.py]: I would remove the default and make this required.

File: examples/nlp/joint_intent_slot_infer_b1.py (hunk 2 above)
Comment (blisc) [reply examples/nlp/joint_intent_slot_infer_b1.py]: I would remove the default and make this required.

**Pull request merged**